test/unit/green-flag.test.jsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-env jest */
+const React = require('react'); // eslint-disable-line no-unused-vars
+const {shallow} = require('enzyme');
+const GreenFlag = require('../../src/containers/green-flag'); // eslint-disable-line no-unused-vars
+const renderer = require('react-test-renderer');
+const VM = require('scratch-vm');
+
+describe('GreenFlag', () => {
+    let vm;
+    beforeEach(() => {
+        // @todo: Ask scratch team their thoughts on mocking the VM.
+        vm = new VM();
+    });
+
+    test('renders active state', () => {
+        const component = renderer.create(
+            <GreenFlag active={true} vm={vm}/>
+        );
+        expect(component.toJSON()).toMatchSnapshot();
+    });
+
+    test('renders inactive state', () => {
+        const component = renderer.create(
+            <GreenFlag active={false} vm={vm}/>
+        );
+        expect(component.toJSON()).toMatchSnapshot();
+    });
+
+    test('triggers onClick when active', () => {
+        const onClick = jest.fn();
+        const componentShallowWrapper = shallow(
+            <GreenFlag active={true} onClick={onClick} vm={vm}/>
```

Comment:
After working with this some more, I think where possible, we should use enzyme to do dom assertions because snapshot testing is:

- not declarative of what you're testing
- prone to human error because you have to look at a long dom snapshot

For example, with the sprite selector item component, the dom is different depending on whether it's in a selected state, unselected state and if a costumeURL property is passed. I could have tested this with the snapshot (which I did try) but I found using the enzyme tools much more declarative. e.g.

```
    test('does not have a close box when not selected', () => {
        selected = false;
        const componentShallowWrapper = shallow(getComponent());
        // this is very explicitly checking for a close button
        expect(componentShallowWrapper.find(CloseButton).exists()).toBe(false);
    });
```

vs this where i have to go look at the snapshot file to confirm i got what i expected (which is prone to human error)
```
    test('does not have a close box when not selected', () => {
        selected = false;
        const component = renderer.create(getComponent());
        // there's nothing here confirming that the close box is or isn't there
        expect(component.toJSON()).toMatchSnapshot();
    });
```